### PR TITLE
PySMO: fix warnings

### PR DESCRIPTION
**Pull request by OOAmusat**
## Fixes
#1403, #1363 

## Summary/Motivation:
Recent changes to the numpy backend have resulted in a large number of deprecation warnings (>7500) being raised in PySMO. This PR makes changes to address these warnings.

## Changes proposed in this PR:
- Fix all numpy deprecation warnings, primarily arising from assigning scalar values to arrays, e.g., `y[i, 0] = x`. This is resolved in this PR by either extracting the specific elements of the scalar (`y[i, 0] = x[0]`) or using the item method (`y[i, 0] = x.item()`)
- Replacing `warnings.warn` with the idaes logger warnings, and fix tests to use `caplog` to reflect this change
- Adding  `@pytest.mark.filterwarnings` to specific tests to ignore runtime warnings around `np.log`

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/surrogate/pysmo/tests/test_kriging.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def test__init__07(self, array_type):
 
     @pytest.mark.unit
     @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    @pytest.mark.filterwarnings("ignore::RuntimeWarning")
```
Comment (lbianchi-lbl) [inline idaes/core/surrogate/pysmo/tests/test_kriging.py:100]: Do you have a list of the `RuntimeWarning`s that are being filtered here?

Comment (OOAmusat) [reply idaes/core/surrogate/pysmo/tests/test_kriging.py:100]: I am filtering two types of runtime warnings:

`RuntimeWarning: invalid value encountered in log
RuntimeWarning: divide by zero encountered in log`
  
However, I'm happy to remove the filters - the runtime warnings are few (about 10 of them). I'm on the fence about whether to filter them or not.

Comment (lbianchi-lbl) [reply idaes/core/surrogate/pysmo/tests/test_kriging.py:100]: I see, thanks. If I understand how `pytest.mark.filterwarnings` works, I think you should be able to use a regular expression to catch both of those cases:

```suggestion
    @pytest.mark.filterwarnings("ignore:(?:invalid value|divide by zero) encountered in log:RuntimeWarning")
```

The goal that we're aiming to is to have no warnings at all that are not expected, so `pytest.mark.filterwarnings` is a good approach, as long as the filter is as specific is possible. In this case, for example, not adding a regex would let possible future `RuntimeWarning` being raised go undetected (which we don't want).

Comment (OOAmusat) [reply idaes/core/surrogate/pysmo/tests/test_kriging.py:100]: Oh, I didn't know I could do that. 

I agree on the "specific" bit - that is why I was on the fence about adding the filters to start with. 

I'll make the change and see it it works. 

Thanks!

Comment (OOAmusat) [reply idaes/core/surrogate/pysmo/tests/test_kriging.py]: Resolved

**Review by lbianchi-lbl — APPROVED** [review]
Thanks a lot @OOAmusat. From a quick look in the pytest job output log, I believe this resolves all of the unanticipated warnings from the PySMO submodules (i.e., I wasn't able to spot any).

The only one that looks like it might be related is the following:

image

However, if you feel this belongs outside of PySMO's "jurisdiction", then feel free to disregard and we can address it at a later time.

File: idaes/core/surrogate/pysmo/tests/test_kriging.py [hunk 2]
```diff
@@ -97,6 +97,9 @@ def test__init__07(self, array_type):
 
     @pytest.mark.unit
     @pytest.mark.parametrize("array_type", [np.array, pd.DataFrame])
+    @pytest.mark.filterwarnings(
+        "ignore:invalid value encountered in log:RuntimeWarning"
```
Comment (andrewlee94) [inline idaes/core/surrogate/pysmo/tests/test_kriging.py:101]: Just for my understanding, where are these warnings coming from and are we sure that they do not indicate an underlying issue somewhere?

**Review by andrewlee94 — APPROVED** [review]
This looks good to me. My only questions is just to confirm that the warnings about invalid values in logs is not a sign of underlying issues.

File: idaes/core/surrogate/pysmo/tests/test_kriging.py (hunk 2 above)
Comment (OOAmusat) [reply idaes/core/surrogate/pysmo/tests/test_kriging.py:101]: The warnings are coming from zeros and NaNs when computing logs. The specific warning is coming from this computation:

```
 RuntimeWarning: invalid value encountered in log:
 log_like = (0.5 * ns * np.log(ssd)) + (0.5 * lndetcov)
```
The warnings are expected: ssd and lndetcov involve matrix manipulations (e.g. multiplication, diagonalization, cholesky factorization), and in some cases that may lead to NaNs or zeros during computation.

**Pull request merged**